### DISCUSSION TITLE
Job Encryption

### DISCUSSION THREAD

**Discussion by AlexJump24** [enlightn/enlightn#51] [Ideas]
Hey,

Following the Laravel changelog, and reading Mohamed's latest blog post https://divinglaravel.com/job-encryption-in-laravel I saw that Encryption capabilities have been added to Laravel 8. This is awesome and we've had to roll some custom solutions for this in past.

Would this be something that Enlightn could potentially add a check for in the future for Laravel 8, given there is next to no overhead of adding it to jobs https://github.com/laravel/framework/pull/35527.

This could mean as a security practice that Enlightn could warn users that the interface should be added to jobs, so the risk of personal information, passwords etc cannot be leaked via the database.

Thanks for all of your great work as ever and excited to hear your thoughts on this.

Alex

**Comment by paras-malhotra** [comment]
Hi @AlexJump24, thanks for posting this idea.

I feel like there are many ways to bell the cat. [Encrypted casts](https://github.com/laravel/framework/pull/34937) work fine for sensitive fields like personal info. I feel as though while encrypted jobs are a great feature, I don't think they should be enforced for every job.

There are downsides on performance and payload size to using encryption all the time. I know the PR you linked says the performance impact is minimal but I feel 50 ms for a 2 MB payload is not small. 

I think this should be specific to the use case. If you're already using encrypted casts for sensitive fields, there is no reason to use encrypted jobs. 

Also, the `SerializesModels` trait only adds the ID of the model to the payload. So, if most of your jobs only pass models in, there is no security risk because the model data is not added to the payload. That's why most jobs would actually not need to be encrypted imo. Mohamed's blog example is a good one that needs to be encrypted because the social security number is passed directly and not as a model but I think that's a rare case.

Would love to hear your thoughts on this.

**Reply by AlexJump24** [reply]
Hi @paras-malhotra,

Thanks for the reply and yeah what you have detailed makes a lot of sense and has given valuable insight into a few things I wasn't aware of. 😃 

A lot of the time I don't pass models directly into jobs and instead make use of data transfer object classes where I don't have the functionality that eloquent provides. However I appreciate that most do within the Laravel community and based on your reasoning I agree it wouldn't make sense to enforce this.

Alex
